board.rb

```diff
@@ -1,89 +1,74 @@
 class Board
-  attr_reader :person, :level
-
-  PERSON  = "@"
-  CRATE   = "o"
-  WALL    = "#"
-  OPEN    = " "
-  STORAGE = "."
-  PERSON_ON_STORAGE = "+"
-  CRATE_ON_STORAGE  = "*"
+  attr_reader :person, :level, :cells
 
   def initialize(level:)
     @level = level
     @person = Person.new(self)
-    make_board
+    create_board
   end
 
-  def make_board
+  def create_board
     lines = File.readlines("sokoban_levels.txt")
     levels = lines.map { |line| line.chomp.ljust(19) }.join("\n")
     levels = levels.split(/\n {19}\n/)
 
-    @cells = levels[level - 1]
+    @cells = levels[level - 1].chars.map { |char| Cell.new(self, char) }
   end
 
   def to_s
-    cells
+    board_cells = cells.map {|c| c.occupant}
+    board_cells.join
+  end
+
+  def cell_at(index)
+    cells[index]
   end
 
   def update(index)
-    new_location = person.location + index
-    if available?(new_location)
+    new_location = cell_at(person.location + index)
+    new_crate_location = cell_at(person.location + (index * 2))
+    if new_location.available?
       move_person(new_location)
-    elsif crate?(new_location) && available?(new_location + index)
+    elsif (new_location.crate? || new_location.crate_on_storage?) && new_crate_location.available?
       move_person(new_location)
       move_crate(index)
     end
   end
 
   def level_completed?
-    !cells.include?("o")
+    cells.none? {|c| c.crate? }
```

Comment:
Space between { and | missing.<br>Pass `&:crate?` as an argument to `none?` instead of a block.